client/js/lounge.js

```diff
@@ -52,6 +52,84 @@ $(function() {
 
 	var favicon = $("#favicon");
 
+	var emoji = new window.EmojiConvertor();
+	emoji.replace_mode = "unified";
+	emoji.replace_colons("");  // this is needed to create the colons map
+	var emojies = Object.keys(emoji.map.colons);
+
+	// Autocompletion Strategies
+
+	var emojiStrategy = {
+		id: "emoji",
+		match: /\B:([-+\w]*)$/,
+		search: function(term, callback) {
+			callback($.map(emojies, function(e) {
+				return e.indexOf(term) === 0 ? e : null;
+			}));
+		},
+		template: function(value) {
+			return emoji.replace_colons(":" + value + ":") + " " + value;
+		},
+		replace: function(value) {
+			return emoji.replace_colons(":" + value + ":") + " ";
+		},
+		index: 1
+	};
+
+	var nicksStrategy = {
+		id: "nicks",
+		match: /\B@([a-zA-Z_[\]\\^{}|`@][a-zA-Z0-9_[\]\\^{}|`-]*)$/,
```

Comment:
I think we need to still allow tabcomplete without the @ (so this should work as it currently does, but if I just type "ya<tab>" it should autocomplete with "yamanickill" as that's how a lot of people use it.